### PR TITLE
feat: search by example drag and drop

### LINKED ISSUE #739: The user can search by example in the Data explorer
It looks like the drag 'n drop won't make it to the Hackathon/demo by early Feb 2023. Alternatively, the user would have two ways to perform a search-by-example: 1) by opening the menu of any card added in the explorer's cart and selecting an option to do a direct search by example (bypassing the search-by-example modal). 2) by clicking a dedicated button on any search results' cards to issue a search by example command, which will (open the modal and) automatically add the record to the search-by-example area.

Related issues:
https://github.com/DARPA-ASKEM/TERArium/issues/446

Design Slides:
Image

Image

Figma URL:
https://www.figma.com/file/wIsOK4hNHc3jxEi69HDNPK/TERArium-HMI-Designs?node-id=8222%3A145867&t=AVcYHkE8nxOwb0Sf-4

### PR DESCRIPTION
**Pull request by echl**
https://user-images.githubusercontent.com/16358728/222500390-7543b896-8d0d-4671-a4b5-6d5dbbc61b04.mov

https://user-images.githubusercontent.com/16358728/222500779-e9201b53-0e70-43ac-bb62-94516c9ddcaa.mov

Resolves #739

### CODE REVIEW

File: packages/client/hmi-client/package.json [hunk 1]
```diff
@@ -32,7 +32,7 @@
 		"pixi.js": "7.0.5",
 		"primeflex": "3.3.0",
 		"primeicons": "6.0.1",
-		"primevue": "3.21.0",
+		"primevue": "^3.23.0",
```
Comment (YohannParis) [inline packages/client/hmi-client/package.json:35]: `yarn add --exact primevue`

File: packages/client/hmi-client/src/utils/drag-drop.ts [hunk 2]
```diff
@@ -0,0 +1,12 @@
+import { ref } from 'vue';
```
Comment (YohannParis) [inline packages/client/hmi-client/src/utils/drag-drop.ts:1]: Adding a commnet explaining the purpose of this composable.
Add MDN links to explain to the Map.

File: packages/client/hmi-client/src/page/data-explorer/search-by-example.ts [hunk 3]
```diff
@@ -0,0 +1,13 @@
+import { ref } from 'vue';
+import { ResultType, SearchByExampleOptions } from '@/types/common';
+
+const searchByExampleOptions = ref<SearchByExampleOptions>({
+	similarContent: false,
```
Comment (YohannParis) [inline packages/client/hmi-client/src/page/data-explorer/search-by-example.ts:5]: Make this `true`

File: packages/client/hmi-client/src/page/data-explorer/components/search-bar.vue [hunk 4]
```diff
@@ -27,27 +27,103 @@
...
 				<i class="pi pi-times clear-search" :class="{ hidden: !query }" @click="clearQuery" />
 			</span>
-			<!-- <i class="pi pi-history" title="Search history" /> -->
-			<!-- <i class="pi pi-image" title="Search by Example" @click="toggleSearchByExample" /> -->
+			<Button
+				class="p-button-secondary search-by-example-button"
+				icon="pi pi-file"
+				@click="isSearchByExampleVisible = !isSearchByExampleVisible"
+				:active="isSearchByExampleVisible"
+			></Button>
```
Comment (YohannParis) [inline packages/client/hmi-client/src/page/data-explorer/components/search-bar.vue:35]:
```suggestion
			/>
```

**Review by YohannParis — APPROVED** [review]
LGTM 
Remove all comment and console

**Review by YohannParis — APPROVED** [review]
Beautiful 🎉

**Comment by mwdchang** [comment]
Slight preference if we move `drag-drop.ts` out of src/util as it is a very Vue-specific kind of mechanism.

**Comment by echl** [comment]
> Slight preference if we move `drag-drop.ts` out of src/util as it is a very Vue-specific kind of mechanism.

What do you suggest?

**Review by YohannParis — APPROVED** [review]

**Pull request merged**